content/articles/push-notification-in-nodejs-using-service-worker/index.md

```diff
@@ -0,0 +1,243 @@
+### Introduction
+A push notification is a message that pops on the screen when a mobile or a web application is performing another activity. In modern web applications, push notifications have a wider place. Push notifications engage the users and draw their attention to a new activity occurring in the web application even when they are not using the application.
+
+The benefits of push notifications include:
+- Increased application engagement as users get engaged even when they are not using the app.
+- Boost user retention by sending them personalized offers and advertisements.
+- Track user metrics to determine the level of success of an application.
+- Enhance the user experience by conveniently providing vital information to the user.
+
+### Goal
+In this article, I will explain a step-by-step implementation of push notifications using service workers in a Nodejs application.
+
+### Table of contents
+- Introduction
+- Prerequisites
+- Creating the node application
+- Building the server-side
+- Coding the client-side
+- Conclusion & Further Reading

```

Comment:
Consider changing your table of content to:-

### Table of contents
- [Introduction](#introduction)
- [Goal](#goal)
- [Table of contents](#table-of-contents)
- [Prerequisites](#prerequisites)
- [Creating the Nodejs Application](#creating-the-nodejs-application)
  - [Init application](#init-application)
  - [Install dependencies](#install-dependencies)
- [Building the server side](#building-the-server-side)
  - [Import the dependencies](#import-the-dependencies)
  - [Generate VAPID Keys](#generate-vapid-keys)
  - [Setting the VAPID Keys with web push](#setting-the-vapid-keys-with-web-push)
  - [Creating the subscribe route](#creating-the-subscribe-route)
- [Coding the client side](#coding-the-client-side)
  - [Setting the static path](#setting-the-static-path)
  - [Creating the required files](#creating-the-required-files)
  - [Setting up the service worker](#setting-up-the-service-worker)
- [Run the server](#run-the-server)
- [Conclusion  and further reading](#conclusion--and-further-reading)